storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -2643,7 +2643,7 @@ dberr_t fseg_free_page(fseg_header_t *seg_header, fil_space_t *space,
     mtr->x_lock_space(space);
 
   DBUG_PRINT("fseg_free_page",
-             ("space_id: " ULINTPF ", page_no: %u", space->id, offset));
+             ("space_id: %" PRIu32 ", page_no: %" PRIu32, space->id, offset));
```

Comment:
I found `ULINTPF` here:
https://github.com/MariaDB/server/blob/25b5c63905682a0e38d8ed86170d98be6ddb8cf8/storage/innobase/include/univ.i#L290-L321
The file even has platform-detecting macros for `UINT32PF`. Is there a reason not to use [`PRIu32` from C/C++](https://en.cppreference.com/w/cpp/header/cinttypes#Macros) (as with `my_snprintf`)? If not, I prefer utilizing the standard.
(also applies to the `PRIu64` addition on `storage/innobase/handler/ha_innodb.cc`)
See also [Zulip discusson on C/C++ `stdint.h` vs. MariaDB `my_global.h`](https://mariadb.zulipchat.com/#narrow/stream/118759-general/topic/stdint.2Eh.20vs.2E.20my_global.2Eh)

It’s also intriguing that `innobase` has this _misleading_ `#define ULINTPF "%zu"` macro *at home* – `%zu` takes an unsigned `size_t`, which is not necessarily an `unsigned long`.
Regardless, looks like `space->id` is an `uint32_t` (not even an `ib_uint32_t` from above).
https://github.com/MariaDB/server/blob/25b5c63905682a0e38d8ed86170d98be6ddb8cf8/storage/innobase/include/fil0fil.h#L328

Reply:
And now `%zu` has backfired in `storage/innobase/dict/dict0load.cc`.

Reply:
@ParadoxV5 I think that we should remove the unnecessary `ifdef` mess if MSVC, Apple Xcode and IBM AIX are more standard compliant nowadays. This used to be an even bigger mess, before someone suggested that we can simply use `typedef size_t ulint` and `"%zu"` instead of performing preprocessor tricks to try to determine the word size and then choose some non-standard names for 64-bit or 32-bit types.

I think that any `" ULINTPF "` can be replaced with `%zu` for the sake of readability. If we can get rid of the above mentioned `ifdef` mess, then we could also replace any `" UINT32PF "`  with `%" PRIu32 "`, but I am not sure which one is better.

I see no reason to retain any `ib_uint32_t` or `ib_uint64_t` types. They should be replaced globally with the standard types. But this should be a separate commit, possibly also a separate pull request.

It is worth noting that until ca501ffb04246dcaa1f1d433d916d8436e30602e we used `ulint` for some things that use only 32 bits for persistent storage. It would make sense to apply that fix as well as any follow-up fixes (such as fea8375893dcb400e0eb6ef78f30cb01b9edc19f) to 10.6, to avoid unnecessary differences between major versions. The 10.5 branch does not matter that much; it will reach end of life rather soon.

Reply:
Perhaps it is time we raise awareness of our unnecessary in-house macros (including [“stdint.h vs. my_global.h” on Zulip](https://mariadb.zulipchat.com/#narrow/channel/118759-general/topic/stdint.2Eh.20vs.2E.20my_global.2Eh)) in a (`beginner-friendly`?) JIRA ticket.

Fixed-width integers, platform-agnostic `printf` macros for them, and additional `printf` size specifiers were standardized in C99 and C++11, which are [our coding standard](https://github.com/MariaDB/server/blob/e6df06d40d/CODING_STANDARDS.md#language-standards) since at least 77c184df7c.

> I think that any `" ULINTPF "` can be replaced with `%zu` for the sake of readability.

Perhaps you’ve typoed. `%zu` is for unsigned `size_t`, which is not necessarily equal to `unsigned long`. The correct standardized specifier is `%lu`.

(P.S.: [This StackOverflow response](https://stackoverflow.com/a/15637571) says they’re different on Windows 64.)

> If we can get rid of the above mentioned `ifdef` mess, then we could also replace any `" UINT32PF "` with `%" PRIu32 "`, but I am not sure which one is better.

`PRIu32` is the standardized macro for unsigned 32-bits (`uint32_t`).

Reply:
Actually, `ulint` is (misleadingly) defines as `size_t` just above the block in thread-top comment:
https://github.com/MariaDB/server/blob/25b5c63905682a0e38d8ed86170d98be6ddb8cf8/storage/innobase/include/univ.i#L285-L288

I’ve searched JIRA and found that it originated from https://github.com/MariaDB/server/commit/5684aa220cc4415fe8ee31ac044210b9ff8c7ce1#diff-64339b1a7c9eaa072f8fd2aa3dd849b382ef3657a7f738b4a79cc4c7026ca81e ([MDEV-12488](https://jira.mariadb.org/browse/MDEV-12488))

Reply:
> I see no reason to retain any `ib_uint32_t` or `ib_uint64_t` types. They should be replaced globally with the standard types. But this should be a separate commit, possibly also a separate pull request.

I have separately filed [MDEV-35460](https://jira.mariadb.org/browse/MDEV-35460) for these matter.